web_responsive/static/src/css/web_responsive.scss

```
@@ -324,19 +344,67 @@ html .o_web_client .o_main .o_main_content {
 .o_content, .modal-content {
     max-width: 100%;
 
+    // Sticky Header & Footer in List View
+    >div {
+        >.table-responsive {
+            overflow-x: initial;
+
+            >.o_list_view {
+                // th & td are here for compatibility with chrome
+                thead, thead tr:nth-child(1) th {
+                    position: sticky;
+                    top: 0;
+                    z-index: 1;
+                }
+                thead tr:nth-child(1) th {
+                    background-color: $o-list-footer-bg-color;
+                }
+                tfoot, tfoot tr:nth-child(1) td {
+                    position: sticky;
+                    bottom: 0;
+                }
+                tfoot tr:nth-child(1) td {
+                    background-color: $o-list-footer-bg-color;
+                }
+            }
+        }
+    }
+
     // Form views
     .o_form_view {
         .o_form_sheet {
             max-width: calc(100% - 32px);
         }
 
+        // Sticky statusbar
+        .o_form_statusbar {
+            position: sticky;
+            top: 0;
+            z-index: 1;
+        }
+
+        // Support for long title (with ellipsis)
+        .oe_title {
+            width: initial;
+
+            >h1>span {
+                max-width: 100%;
+                text-overflow: ellipsis;
+                white-space: nowrap;
+                overflow: hidden;
+                width: initial;
+            }
+            >h1>span:active {
```

Here and above, the same about tight structure. Can't we be less strict?

Also a suggestion about readability (this applies to your whole patch, BTW):

```suggestion
            > h1 > span:active {
```